pkg/driver/nydus/nydus.go

```diff
@@ -224,13 +229,40 @@ func (d *Driver) Convert(ctx context.Context, provider accelcontent.Provider, so
 	if err != nil {
 		return nil, errors.Wrap(err, "get source image")
 	}
+
+	cacheRef := provider.SetCacheRef("")
+	useRemoteCache := cacheRef != ""
+	if useRemoteCache {
+		logrus.Infof("remote cache image reference: %s", cacheRef)
+		if err := d.FetchRemoteCache(ctx, provider, cacheRef); err != nil {
+			if errors.Is(err, errdefs.ErrNotSupport) {
+				logrus.Warn("Content store does not support remote cache")
+			} else {
+				return nil, errors.Wrap(err, "fetch remote cache")
+			}
+		}
+	}
+
 	desc, err := d.convert(ctx, provider, *image, sourceRef)
 	if err != nil {
 		return nil, err
 	}
 	if d.mergeManifest {
 		return d.makeManifestIndex(ctx, provider.ContentStore(), *image, *desc)
 	}
+
+	if useRemoteCache {
+		// Before update cache, we need to pull cache manifest from remote
```

Comment:
// Fetch the old remote cache before updating and pushing the new one to avoid conflict.

Reply:
fixed.